### PR TITLE
Fix java-quarkus devfile

### PR DESCRIPTION
**Pull request by vinokurig**
### What does this PR do?:
* Update the `schemaVersion` to `2.2.0`
* Add `https` endpoint

### Which issue(s) this PR fixes:
https://github.com/eclipse/che/issues/22869

### PR acceptance criteria:

- [ ] Contributing guide
_Have you read the [devfile registry contributing guide](https://github.com/devfile/registry/blob/main/CONTRIBUTING.md) and followed its instructions?_
- [ ] Test automation
_Does this repository's tests pass with your changes?_
- [ ] Documentation
_Does any documentation need to be updated with your changes?_
- [ ] Check Tools Provider
_Have you tested the changes with existing tools, i.e. Odo, Che, Console? (See [devfile registry contributing guide](https://github.com/devfile/registry/blob/main/CONTRIBUTING.md) on how to test changes)_

### How to test changes / Special notes to the reviewer:

### CODE REVIEW

File: stacks/java-quarkus/devfile.yaml [hunk 1]
```diff
@@ -9,7 +9,7 @@ metadata:
     - Quarkus
   projectType: Quarkus
   language: Java
-  version: 1.3.0
+  version: 1.3.1
```
Comment (michael-valdron) [inline stacks/java-quarkus/devfile.yaml:12]: Hi @vinokurig,

I'd recommend also creating a separate version directory for the new version of the stack, see our `go` stack directory for how this is done: https://github.com/devfile/registry/tree/27951b4afc4feba8fab57538f0ed263d14fa5c18/stacks/go

So the outcome would be a new `stack.yaml` with both `1.3.0/devfile.yaml` and `1.3.1/devfile.yaml` versioned devfiles.

Comment (vinokurig) [reply stacks/java-quarkus/devfile.yaml]: Done

File: stacks/java-quarkus/1.3.1/devfile.yaml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    - Java
+    - Quarkus
+  projectType: Quarkus
+  language: Java
+  version: 1.3.1
+  website: https://quarkus.io
+starterProjects:
+  - name: community
+    zip:
+      location: https://code.quarkus.io/d?e=io.quarkus%3Aquarkus-resteasy&e=io.quarkus%3Aquarkus-micrometer&e=io.quarkus%3Aquarkus-smallrye-health&e=io.quarkus%3Aquarkus-openshift&cn=devfile
```
Comment (vinokurig) [inline stacks/java-quarkus/1.3.1/devfile.yaml:17]: @michael-valdron 
Can we change the project source to the git repository of the [java quarkus sample](https://github.com/devfile-samples/devfile-sample-code-with-quarkus)? That would give an ability to contribute to the sample.

Comment (michael-valdron) [reply stacks/java-quarkus/1.3.1/devfile.yaml:17]: @vinokurig Of course, as the registry repo open a PR with any desired changes to the starter project repo and we can review them.

**Review by michael-valdron — CHANGES_REQUESTED** [review]
Looks like you need to rebase the recent `nodejs-svelte` changes in, this is why odov3 check is failing:
```
 Output from proc 1:
  --- FAIL: TestOdo (0.02s)
      odo_test.go:72: 
          Expected
              <*fs.PathError | 0xc0008f7e30>: 
              stat /home/runner/work/registry/registry/stacks/nodejs-svelte/1.2.1/devfile.yaml: no such file or directory
              {
                  Op: "stat",
                  Path: "/home/runner/work/registry/registry/stacks/nodejs-svelte/1.2.1/devfile.yaml",
                  Err: <syscall.Errno>0x2,
              }
          to be nil
  FAIL

Output from proc 2:
  --- FAIL: TestOdo (0.02s)
      odo_test.go:72: 
          Expected
              <*fs.PathError | 0xc000653650>: 
              stat /home/runner/work/registry/registry/stacks/nodejs-svelte/1.2.1/devfile.yaml: no such file or directory
              {
                  Op: "stat",
                  Path: "/home/runner/work/registry/registry/stacks/nodejs-svelte/1.2.1/devfile.yaml",
                  Err: <syscall.Errno>0x2,
              }
          to be nil
  FAIL
```

**Comment by maxandersen** [comment]
What/where does the Https serving coming from?

How do I test it?

**Comment by vinokurig** [comment]
@michael-valdron 
I merged the upstream main branch into the PR branch and changed the source project from zip to git url, please take a look.

**Comment by vinokurig** [comment]
@maxandersen 
> What/where does the Https serving coming from?
>
> How do I test it?

This is needed to run the application on eclipse CHE. There is a similar PR for .net: https://github.com/devfile/registry/pull/312

**Comment by michael-valdron** [comment]
> > changed the source project from zip to git url
> 
> @vinokurig I misunderstood what you were originally asking, but now I think I do.
> 
> You want to change not just the sample but want to include the sample as the starter project instead of what is generated from https://code.quarkus.io so that the source project can be changed to work with the HTTPS endpoint.
> 
> In the case, I'm not sure if this can be done. Let me discuss with the team and followup later today.
> 
> Apologies for the misunderstanding.
> 
> @maxandersen Let me know if you have any ideas to workaround this, otherwise I'll update you on what devfile services team proposes for this.

@vinokurig Had a discussion with the team and we don't feel you should use the sample as a starter project since we do not maintain it as often as the current starter projects are maintained by the quarkus development team. As @maxandersen is the owner of this stack, I suggest you continue this conversation with them to work out a solution that works best.

**Comment by vinokurig** [comment]
@michael-valdron 
>Had a discussion with the team and we don't feel you should use the sample as a starter project since we do not maintain it as often as the current starter projects are maintained by the quarkus development team. As @maxandersen is the owner of this stack, I suggest you continue this conversation with them to work out a solution that works best.

I reverted the project source item to original.

File: stacks/java-quarkus/1.3.1/devfile.yaml (hunk 2 above)
Comment (vinokurig) [reply stacks/java-quarkus/1.3.1/devfile.yaml:17]: reverted the item

**Comment by maxandersen** [comment]
Is there a problem with the generated Project from code.quarkus ?

**Comment by vinokurig** [comment]
No problems, I just want to add some configuration files for vscode editor.

**Review by michael-valdron — COMMENTED** [review]
If there isn't anything additional needed from the starter projects, the changes looks good to me.

Just need a sign off from @maxandersen.

**Comment by maxandersen** [comment]
Ok. I'm all good with the current changes but I don't see the vscode files you want to add?

**Comment by vinokurig** [comment]
@maxandersen @michael-valdron 
>@vinokurig you could still include them as resource files instead, these would be stored under the registry with the devfile.

Do you have an example of such files? Also I have a question about the `starterProjects` section of the devfile: why there are two zip items with identical content but different source url?

**Comment by maxandersen** [comment]
One is quarkus from community the other red hat product.

**Comment by michael-valdron** [comment]
> Do you have an example of such files?

Our go stack has supporting resources files in the latest version: https://github.com/devfile/registry/tree/72a45de9a3a01834ecbfa4e920926a041622ade6/stacks/go/2.3.0

Any files along with sub-directories under the stack version directory other than the devfile itself will be treated as resource files and extracted with the devfile.

**Review by michael-valdron — CHANGES_REQUESTED** [review]
Going to test odo locally as we are dealing with issues on GitHub workflows using odo with some of the stacks due to reduced resources.

@vinokurig My apologies, one more small thing I suggest to change while I test, since the schema version is being bumped I think the new stack version should be a minor bump so `1.4.0` instead of `1.3.1`.

**Comment by vinokurig** [comment]
@michael-valdron 
> My apologies, one more small thing I suggest to change while I test, since the schema version is being bumped I think the new stack version should be a minor bump so 1.4.0 instead of 1.3.1.

done.

**Comment by vinokurig** [comment]
@michael-valdron @maxandersen 
> Any files along with sub-directories under the stack version directory other than the devfile itself will be treated as resource files and extracted with the devfile.

It is not supported in the Eclipse Che yet. Is there a way to contribute directly to the zip archive of the starter project?

**Review by michael-valdron — COMMENTED** [review]
> Going to test odo locally as we are dealing with issues on GitHub workflows using odo with some of the stacks due to reduced resources.

I ran the tests with odov3 and found that it builds and deploys but during deployment it checks the new endpoint added and there is nothing being served on port 3000, and this is why the odov3 check is failing.

You'll need to have a service that redirects to port 8080 what quarkus is serving through the HTTPS protocol, I'm not sure if this is something that could be done in the devfile itself or needs additional changes to the starter projects from code.quarkus.io and code.quarkus.redhat.com.

odov2 failing check is unrelated as it skips stack versions that use `2.2.x` schema versions.

**Comment by michael-valdron** [comment]
> It is not supported in the Eclipse Che yet. Is there a way to contribute directly to the zip archive of the starter project?

FYI @maxandersen

**Comment by michael-valdron** [comment]
Related to part of https://github.com/devfile/api/issues/1318

**Comment by vinokurig** [comment]
@michael-valdron What if I remove the endpoint with port 300 but set the existed 8080 endpoint as https?

**Comment by vinokurig** [comment]
@michael-valdron @maxandersen Could you please help to figure out why the odo2 test is failing?

**Comment by vinokurig** [comment]
@michael-valdron 
> @vinokurig We can ignore odov2 for this PR since these changes only introduce a stack version that uses schema version 2.2.x which is ignored by this check.

Are we good to go with current changes?

**Review by michael-valdron — APPROVED** [review]
/lgtm

**Review by maxandersen — APPROVED** [review]
dont see an issue but also dont have way to run it atm. so no objection trusting you've tested it works in the various layers.

**Comment by vinokurig** [comment]
@michael-valdron @maxandersen I don't have enough permissions, could you please merge?

**Pull request merged**